addons/barcodes/models/barcode_nomenclature.py

```diff
@@ -141,3 +142,9 @@ def parse_barcode(self, barcode):
                     return parsed_result
 
         return parsed_result
+
+    @staticmethod
+    def check_barcode_value(barcode):
+        regex = "^[\x00-\x7F\xF1-\xF4]+$"
+        if barcode and not re.fullmatch(regex, barcode):
+            raise ValidationError(_("Invalid Barcode Format. Code128 type required."))
```

Comment:
I think it's easier for the user to let they know what is the wrong character
```suggestion
    def check_barcode_is_code128(barcode):
        unsupported_chars = re.findall("[^\x00-\x7F\xF1-\xF4]", barcode)
        if barcode and unsupported_chars:
            error_message = _("Invalid Barcode Format, following characters are not supported: %s")
            raise ValidationError(error_message % " ".join(unsupported_chars))
```
Also, I think the method's name should be clear this check is about Code128 because it exists some other checks for the barcode (the lenght, the checksum)